contracts/strategies/SnowballProxy.sol

```diff
@@ -0,0 +1,122 @@
+    }
+}
+
+contract SnowballProxy {
+    using SafeMath for uint256;
+    using SafeProxy for ISnowballVoter;
+    using SafeERC20 for IERC20;
+
+    uint constant internal BIPS_DIVISOR = 10000;
+
+    uint public SNOB_FEE_BIPS;
+    address constant public SNOB = address(0xC38f41A296A4493Ff429F1238e030924A1542e50);
+    ISnowballVoter public immutable snowballVoter;
+    address public immutable devAddr;
+
+    modifier onlyDev() {
+        require(msg.sender == devAddr, "SnowballProxy::onlyDev");
+        _;
+    }
+
+    modifier onlyStrategy(address _stakingContract) {
+        require(strategies[_stakingContract] == msg.sender, "SnowballProxy:onlyStrategy");
+        _;
+    }
+    
+    // stakingContract => strategies
+    mapping(address => address) public strategies;
+
+    constructor(address _snowballVoter, uint _snobFeeBips) {
+        devAddr = msg.sender;
+        SNOB_FEE_BIPS = _snobFeeBips;
+        snowballVoter = ISnowballVoter(_snowballVoter);
+    }
+
+    function approveStrategy(address _stakingContract, address _strategy) external onlyDev {
+        strategies[_stakingContract] = _strategy;
+    }
+
+    function setSnobFee(uint _snobFeeBips) external onlyDev {
+        SNOB_FEE_BIPS = _snobFeeBips;
+    }
+
+    // Methods for YY Strategies
+    function withdraw(
+        address _stakingContract,
+        address _snowGlobe,
+        address _token,
+        uint amount
+    ) external onlyStrategy(_stakingContract) returns (uint256) {
+        uint256 _balance = IERC20(_token).balanceOf(address(snowballVoter));
+        uint sharesAmount = amount.mul(1e18).div(ISnowGlobe(_snowGlobe).getRatio());
+        snowballVoter.safeExecute(_stakingContract, 0, abi.encodeWithSignature("withdraw(uint256)", sharesAmount));
+        snowballVoter.safeExecute(_snowGlobe, 0, abi.encodeWithSignature("withdraw(uint256)", sharesAmount));
+        _balance = IERC20(_token).balanceOf(address(snowballVoter)).sub(_balance);
+        snowballVoter.safeExecute(_token, 0, abi.encodeWithSignature("transfer(address,uint256)", msg.sender, _balance));
+        return _balance;
+    }
+
+    function withdrawAll(address _stakingContract, address _snowGlobe, address _token) external onlyStrategy(_stakingContract) {
+        snowballVoter.safeExecute(_stakingContract, 0, abi.encodeWithSignature("withdrawAll()"));
+        snowballVoter.safeExecute(_snowGlobe, 0, abi.encodeWithSignature("withdrawAll()"));
+        snowballVoter.safeExecute(_token, 0, abi.encodeWithSignature("transfer(address,uint256)", msg.sender, IERC20(_token).balanceOf(address(snowballVoter))));
+    }
+
+    function deposit(address _stakingContract, address _snowGlobe, address _token) external onlyStrategy(_stakingContract) returns (uint256) {
+        uint256 balance = IERC20(_token).balanceOf(address(this));
+        IERC20(_token).safeTransfer(address(snowballVoter), balance);
+        balance = IERC20(_token).balanceOf(address(snowballVoter));
```

Comment:
we already get balance in L85, so this could be redundant. the only advantage here is to collect dust (if any) in `snowballVoter`. we could improve deposit cost here pending review of dust

Reply:
fixed, didn't leave dust on my tests